### PR TITLE
wallet: Fix sendall with watchonly wallets and specified inputs

### PR DESCRIPTION
**Pull request by achow101**
The `sendall` RPC would previously fail when used with a watchonly wallet and specified inputs. This failure was caused by checking isminetype equality with ISMINE_ALL rather than a bitwise AND as IsMine can never return ISMINE_ALL.

Also added a test.

### CODE REVIEW

**Comment by fanquake** [comment]
Does this need backporting?

**Comment by achow101** [comment]
> Does this need backporting?

Yes

File: src/wallet/rpc/spend.cpp [hunk 1]
```diff
@@ -1379,7 +1379,7 @@ RPCHelpMan sendall()
                         throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Input not available. UTXO (%s:%d) was already spent.", input.prevout.hash.ToString(), input.prevout.n));
                     }
                     const CWalletTx* tx{pwallet->GetWalletTx(input.prevout.hash)};
-                    if (!tx || pwallet->IsMine(tx->tx->vout[input.prevout.n]) != (coin_control.fAllowWatchOnly ? ISMINE_ALL : ISMINE_SPENDABLE)) {
+                    if (!tx || !(pwallet->IsMine(tx->tx->vout[input.prevout.n]) & (coin_control.fAllowWatchOnly ? ISMINE_ALL : ISMINE_SPENDABLE))) {
```
Comment (furszy) [inline src/wallet/rpc/spend.cpp:1382]: Not related to this PR, but.. pretty sure that we can make the node crash by providing an out-of-bounds input prevout index.

e.g. 
```diff
diff --git a/test/functional/wallet_sendall.py b/test/functional/wallet_sendall.py
--- a/test/functional/wallet_sendall.py	(revision d442101d85e1f89b072145345ea0bcf60d6ceacc)
+++ b/test/functional/wallet_sendall.py	(date 1666283169723)
@@ -294,6 +294,7 @@
         else:
             watchonly.importmulti(import_req)
 
+        utxo["vout"] = 10
         sendall_tx_receipt = watchonly.sendall(recipients=[self.remainder_target], options={"inputs": [utxo]})
         psbt = sendall_tx_receipt["psbt"]
         decoded = self.nodes[0].decodepsbt(psbt)
```

**Review by furszy — APPROVED** [review]
Code review ACK d442101d

With an extra crash finding that doesn't necessarily need to be included here.

File: src/wallet/rpc/spend.cpp (hunk 1 above)
Comment (achow101) [reply src/wallet/rpc/spend.cpp:1382]: Fixed and added a test.

**Review by w0xlt — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/26344/commits/315fd4dbabb6b631b755811742a3bdf93e1241bf

Commits can be squashed.

**Comment by achow101** [comment]
> Commits can be squashed.

I prefer to keep implementation separate from test so that tests can be easily cherry-picked onto master to verify that they fail.

**Review by furszy — APPROVED** [review]
ACK 315fd4db

**Comment by fanquake** [comment]
Backported in #26410.

**Pull request merged**